cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreImpl.java

```diff
@@ -1421,77 +982,57 @@ public String extensionDataFolder(String extensionName) {
     public void resolveConflictsForDocument(final String docId, final ConflictResolver resolver)
             throws ConflictException {
 
-        // before starting the tx, get the 'new winner' and see if we need to prepare its
-        // attachments
-
 
         try {
-            queue.submitTransaction(new SQLCallable<Object>() {
-                @Override
-                public Object call(SQLDatabase db) throws Exception {
-                    DocumentRevisionTree docTree = new GetAllRevisionsOfDocumentCallable(docId, attachmentsDir, attachmentStreamFactory).call(db);
-                    if (!docTree.hasConflicts()) {
-                        return null;
-                    }
-                    DocumentRevision newWinner = null;
-                    try {
-                        newWinner = resolver.resolve(docId, docTree.leafRevisions(true));
-                    } catch (Exception e) {
-                        logger.log(Level.SEVERE, "Exception when calling ConflictResolver", e);
-                    }
-                    if (newWinner == null) {
-                        // resolve() threw an exception or returned null, exit early
-                        return null;
-                    }
 
-                    String revIdKeep = newWinner.getRevision();
-                    if (revIdKeep == null) {
-                        throw new IllegalArgumentException("Winning revision must have a revision" +
-                                " id");
-                    }
+            // before starting the tx, get the 'new winner' and see if we need to prepare its
+            // attachments
+            DocumentRevisionTree docTree = queue.submit(new GetAllRevisionsOfDocumentCallable(docId, attachmentsDir, attachmentStreamFactory)).get();
```

Comment:
Oh I was following the comment but yes in the original version it was inside the transaction. It possibly doesn't make much/any difference because `getAllRevisionsOfDocumentCallable` doesn't modify the database, but I could put it back in the transaction to be consistent with the original code.